### PR TITLE
936 block number range 2

### LINKED ISSUE #936: Add a max block number range check
### Problem

Currently it's possible to query the relay for any block number.
However, in some cases this results in a timeout from the mirror node since we continue to poll and never find it e.g. from testnet.

```
[2023-02-22 16:02:39.344 +0000] ERROR (relay-eth/51 on testnet-json-rpc-relay-ff5f7b4f6-bd6kk): [Request ID: 3e792ef6-b9a3-4c06-8077-b55eadcf18a7] Failed to retrieve block for blockNum 0x6e276d0
    err: {
      "type": "MirrorNodeClientError",
      "message": "timeout of 10000ms exceeded",
      "stack":
          Error: timeout of 10000ms exceeded
              at MirrorNodeClient.handleError (/home/node/app/packages/relay/dist/lib/clients/mirrorNodeClient.js:166:15)
              at MirrorNodeClient.<anonymous> (/home/node/app/packages/relay/dist/lib/clients/mirrorNodeClient.js:140:22)
              at Generator.throw (<anonymous>)
              at rejected (/home/node/app/packages/relay/dist/lib/clients/mirrorNodeClient.js:25:65)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      "statusCode": 504
    }
[2023-02-22 16:02:39.344 +0000] ERROR (Internal error/51 on testnet-json-rpc-relay-ff5f7b4f6-bd6kk): [Request ID: 3e792ef6-b9a3-4c06-8077-b55eadcf18a7] [POST] eth_getBlockByNumber: -32603 15343 ms
    code: -32603
    message: "Unknown error invoking RPC"
```

### Solution

Add a max block number range check .

When a request for a block is made and it's a number.
Call the mirror node to get the current block number. If the block number asked for is more than a few blocks greater then we should fail fast.
We should have a reasonable buffer to make sure we don't erroneously fail, noting that current block time is 2 secs

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by ebadiere**
Added check for latest block to avoid regression, multiple calls to the mirrornode.

Fixes #936 

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -1583,7 +1583,7 @@ export class EthImpl implements Eth {
       return null;
     }
   
-    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag)) {
+    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag) && !EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/eth.ts:1586]: Should be a test to confirm behaviour here.
The fact that there's no failing tests means there's missing test coverage

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Nice, but missing test

File: packages/relay/src/lib/eth.ts (hunk 1 above)
Comment (ebadiere) [reply packages/relay/src/lib/eth.ts:1586]: Done.

**Comment by ebadiere** [comment]
Addressed in another PR.

**Pull request closed without merging**